src/app/body/all-products/all-products/all-products.component.ts

```diff
@@ -1,19 +1,23 @@
 import { Component, OnInit } from '@angular/core';
```

Comment:
`src/app/body/all-products/all-products/all-products.component.ts`
Why do you need two same folders `all-products`?

Reply:
Vasyl told me that component should not be without folder with other components, so upper all-products folder is like page includes inner components, the inner one is a component

Reply:
fixed

Reply:
Then page folder should be named with `-page` end:
`src/app/pages/all-products-page/all-products/all-products.component.ts`